### PR TITLE
 Added Proposal endpoint api

### PR DESCRIPTION
**Pull request by jetchirag**
Tested all endpoints

### CODE REVIEW

**Comment by jetchirag** [comment]
Postman json: https://www.getpostman.com/collections/1bbeb57173f892662736

File: src/models/Proposal.js [hunk 1]
```diff
@@ -5,14 +5,17 @@ const Proposal = mongoose.model('Proposal', new mongoose.Schema({
...
 	clubId: {
 		type: mongoose.Schema.Types.ObjectId,
-		ref: 'Club',
+		ref: 'Club', 
+		// TODO: Should this be immutable 
 	},
 	facultyId: {
 		type: mongoose.Schema.Types.ObjectId,
 		ref: 'FC'
+		// TODO: Should this be immutable
```
Comment (ayushpanwar25) [inline src/models/Proposal.js]: yes all 3 immutable

File: src/routes/club/Proposal.js [hunk 2]
```diff
@@ -28,14 +34,80 @@ router.get("/get/:id", verifyClub, async (req, res) => {
...
+				})
+		})
+		.catch((err) => {
+			res.status(404).send({ success: false, message: "Cannot find club ID" })
+		});
+
+})
 
 router.put("/edit/:id", verifyClub, async (req, res) => {
+	// TODO: Add a check so club cannot edit major or any data once it's approved by faculty
```
Comment (ayushpanwar25) [inline src/routes/club/Proposal.js:61]: I think that clubs should be able to store incomplete proposals which are not forwarded to any faculty so they can work on it over time. We shouldn't allow clubs to edit proposals at all after forwarding.

File: src/routes/club/Proposal.js [hunk 3]
```diff
@@ -28,14 +34,80 @@ router.get("/get/:id", verifyClub, async (req, res) => {
...
+			}
+		})
+		.catch((err) => {
+			res.status(404).send({ success: false, message: "Proposal not found", error: err })
+		})
 
 });
 
 router.delete("/delete/:id", verifyClub, async (req, res) => {
+	// TODO: Should clubs be allowed to simply delete a propoasl
```
Comment (ayushpanwar25) [inline src/routes/club/Proposal.js:88]: Will need input from club heads.

File: src/routes/club/Proposal.js (hunk 2 above)
Comment (jetchirag) [reply src/routes/club/Proposal.js:61]: Agreed. Will add a approval == 0 check. ~~Maybe we can also add a submit or ready for approval field~~

File: src/models/Proposal.js (hunk 1 above)
Comment (jetchirag) [reply src/models/Proposal.js]: Done

**Pull request merged**